### PR TITLE
Fix nightly CI

### PR DESCRIPTION
**Pull request by sequencer**
Cherry picked #3662 and #3659:

- [binder] Update firtool-lib C-API to allow the creation of multiple options
- [temp] Try to fix the CI script
- [circt] Use new public-modules option

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -14,6 +14,7 @@ test_run_dir
 *~
 \#*\#
 .\#*
+.vscode
```
Comment (sequencer) [inline .gitignore:17]: not related

File: .github/workflows/ci-circt-nightly.yml [hunk 2]
```diff
@@ -34,7 +34,7 @@ jobs:
         id: determine-branch
         shell: bash
         run: |
-          branch=$(git rev-parse --abbrev-ref HEAD)
+          branch=$(git rev-parse HEAD)
```
Comment (sequencer) [inline .github/workflows/ci-circt-nightly.yml:37]: do I need it

**Comment by sequencer** [comment]
(seems nightly is real **nightly**, wait nightly release tonight.)

File: .github/workflows/ci-circt-nightly.yml [hunk 3]
```diff
@@ -24,7 +24,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Update Staging Branch
-        uses: circt/update-staging-branch@v1.1.0
+        uses: SpriteOvO/update-staging-branch@fix-for-fork
```
[the remark this answers is not in the document]
Comment (seldridge) [reply .github/workflows/ci-circt-nightly.yml:27]: `circt/update-staging-branch` has a `v1.2.0` release that can be used now. Thanks for the PR @SpriteOvO. 😉

**Pull request merged**